evm_arithmetization/src/prover.rs

```diff
@@ -522,6 +523,92 @@ fn build_segment_data<F: RichField>(
     }
 }
 
+pub struct SegmentDataIterator {
+    pub partial_next_data: Option<GenerationSegmentData>,
+    pub inputs: GenerationInputs,
+    pub max_cpu_len_log: Option<usize>,
+    pub nb_segments: usize,
```

Comment:
Why do we have this? It seems unused, and mimics the `segment_idx` that's already in `partial_next_data`